src/dds/rs-dds-device-proxy.cpp

```diff
@@ -136,17 +136,8 @@ dds_device_proxy::dds_device_proxy( std::shared_ptr< context > ctx, std::shared_
         int sensor_index = 0;
     };
     std::map< std::string, sensor_info > sensor_name_to_info;
-    // We assign (sid,index) based on the stream type:
-    // LibRS assigns streams names based on the type followed by an index if it's not 0.
-    // I.e., sid is based on the type, and index is 0 unless there's more than 1 in which case it's 1-based.
-    int counts[RS2_STREAM_COUNT] = { 0 };
-    // Count how many streams per type
-    _dds_dev->foreach_stream( [&]( std::shared_ptr< realdds::dds_stream > const & stream )
-                              { ++counts[to_rs2_stream_type( stream->type_string() )]; } );
-    // Anything that's more than 1 stream starts the count at 1, otherwise 0
-    for( int & count : counts )
-        count = ( count > 1 ) ? 1 : 0;
-    // Now we can finally assign (sid,index):
+    std::map< sid_index, sid_index > type_and_index_to_dds_stream_sidx;
```

Comment:
I'm not entirely sure what you're doing here... Maybe an explanation is in order?

Reply:
Added